redis/cluster.py

```diff
@@ -455,22 +459,29 @@ def __init__(
             When set to True: all slots must be covered to construct the
             cluster client. If not all slots are covered, RedisClusterException
             will be thrown.
-        :read_from_replicas: 'bool'
+        :param read_from_replicas:
              Enable read from replicas in READONLY mode. You can read possibly
              stale data.
              When set to true, read commands will be assigned between the
              primary and its replications in a Round-Robin manner.
-         :dynamic_startup_nodes: 'bool'
+         :param dynamic_startup_nodes:
              Set the RedisCluster's startup nodes to all of the discovered nodes.
              If true (default value), the cluster's discovered nodes will be used to
              determine the cluster nodes-slots mapping in the next topology refresh.
              It will remove the initial passed startup nodes if their endpoints aren't
              listed in the CLUSTER SLOTS output.
              If you use dynamic DNS endpoints for startup nodes but CLUSTER SLOTS lists
              specific IP addresses, it is best to set it to false.
-        :cluster_error_retry_attempts: 'int'
-             Retry command execution attempts when encountering ClusterDownError
-             or ConnectionError
+        :param cluster_error_retry_attempts:
+             Number of times to retry before raising an error when
+             :class:`~.TimeoutError` or :class:`~.ConnectionError` or
+             :class:`~.ClusterDownError` are encountered
+        :param connection_error_retry_attempts:
+            Number of times to retry before reinitializing when :class:`~.TimeoutError`
+            or :class:`~.ConnectionError` are encountered.
+            The default backoff strategy will be set if Retry object is not passed (see
+            get_default_backoff in backoff.py). To change it, pass a custom Retry object
+            using the "retry" keyword.
         :reinitialize_steps: 'int'
```

Comment:
I think 10 is too many - but yes I agree that 1/3 is too small. What in your experience leads you to 10?

Reply:
There's a latency tradeoff between getting a moved error and rotating the request to another node, which will result in 2 call rounds, and between reinitializing the cluster on a moved error which is an expensive operation that will also require another call to the redirected node. As a new cluster map can take some time to form (e.g. when adding more shards to the cluster, when migrating slots from one node to another), we can receive lots of moved errors during this time. By setting the reinitiailize steps to 10, we will not have to call the expensive reinitialize call on every moved error, giving the cluster more time to reach the final slot map before querying for it. The slot map will be changed at once in other scenarios, such as when the primary fails and a failover is done, so reinitializing the cluster map after the first Moved error should suffice, and if we did indeed query for the cluster slot map after the first move, we could potentially save 9 more moved errors.
So, I don't think we have a best value that fits all cases. Some other clients reinitialize the topology for each moved error and some continue to change only the specific slot owner when moved occurred.
We can reduce it to 5 (or less) if you feel 10 is too much, either way some cases will benefit from either one.

Reply:
I don't think there's a right answer here, without data. Given that they can be overriden - that's part of the answer. I'd rather allow for a lower retry threshold (5), while allowing users to override. Reset to 5 seems reasonable - but again, I have no measured, scientific reason to dictate any number

Reply:
changed to 5.